matrix-test.yaml

```
@@ -4,8 +4,12 @@ pull-request:
   - { CUDA_VER: '11.8', ARCH: 'amd64', PYTHON_VER: '3.9', GPU: 'v100', DRIVER: 'earliest' }
   - { CUDA_VER: '12.0', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
   - { CUDA_VER: '12.0', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
+  - { CUDA_VER: '12.2', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
+  - { CUDA_VER: '12.2', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```

I think we should keep the PR matrix small, so just update one of the 12.0 to 12.2 basically.

```suggestion
  - { CUDA_VER: '12.0', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
  - { CUDA_VER: '12.2', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
```

---

To keep the PR matrix small, I would modify the last line here (amd64 CUDA 12.2) to use Python 3.11 when we're ready for that. Maybe add Python 3.11 for one of the ARM jobs too.

---

Thinking about this more, the CUDA 12.2 ARM images have more utility than the CUDA 12.0 ARM images. For example cuFile is supported in CUDA 12.2+ (though not earlier). So what if we flipped these?

```suggestion
  - { CUDA_VER: '12.0', ARCH: 'amd64', PYTHON_VER: '3.10', GPU: 'v100', DRIVER: 'latest' }
  - { CUDA_VER: '12.2', ARCH: 'arm64', PYTHON_VER: '3.10', GPU: 'a100', DRIVER: 'latest' }
```

---

Committing the last change. Happy to discuss further on the updated diff